### PR TITLE
meta: update next.js and remove robots.txt gen

### PR DESCRIPTION
**Pull request by ovflowd**
This PR updates Next.js to v13.5.1 and removes the Sitemap CLI generation in favor of a static `robots.txt` file (as it never changes)

### CODE REVIEW

**Comment by ovflowd** [comment]
Blocked by https://github.com/storybookjs/storybook/issues/24234

File: public/robots.txt
Comment (AugustinMauroy) [inline public/robots.txt]: we probably should use [robot api](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/robots). WDYT?

Comment (ovflowd) [reply public/robots.txt]: We really don't need to, but that sounds a good alternative. Weird that if you search "robots.txt next" first result is https://nextjs.org/learn/seo/crawling-and-indexing/robots-txt which still has no reference to the new Api.

Good find!

Comment (AugustinMauroy) [reply public/robots.txt]: Personally I use it because it's an advantage to have everything in nextjs under the app dir. _after that it's a personal opinion_.

File: app/robots.ts
Comment (AugustinMauroy) [inline app/robots.ts]:
```suggestion
sitemap: `${baseUrlAndPath}/sitemap.xml`,
host: 'https://nodejs.org'
});
```
We should add host ??

Comment (ovflowd) [reply app/robots.ts]: I don't think so; it's optional and not mandatory by the robots.txt spec

Comment (AugustinMauroy) [reply app/robots.ts]: actually on we have robot entry. So I'm ok to remove it but keep it's cost nothing and it's can useful.
REF: https://nodejs.org/robots.txt

Comment (ovflowd) [reply app/robots.ts]: We do have, but it has no value. It is there because the `next-sitemap` script required it.

I don't see any benefit in having it. It has no value for us, and the search engines already know the host. No website that I know uses that field.

File: app/robots.ts
Comment (bmuenzenmeyer) [inline app/robots.ts]:
```suggestion
// This allows us to generate a `robots.txt` file dynamically based on the needs of the Node.js Website
```

**Review by bmuenzenmeyer — CHANGES_REQUESTED** [review]
one typo

this still looks really odd to me that in a PR where we are excited about new next functionality, our package.json is set to `13.4.19`

File: app/robots.ts
Comment (bmuenzenmeyer) [reply app/robots.ts]: gosh the docs are pretty sparse around sitemap... next links to wikipedia... which links to a yahoo article from 2009

**Comment by ovflowd** [comment]
> this still looks really odd to me that in a PR where we are excited about new next functionality, our package.json is set to 13.4.19

Why is this a typo? The package-lock is what matters.

**Comment by yannbf** [comment]
Hey everyone! Thanks for reporting this, we know about the regression and are currently working on a fix for it. We will release it as soon as we can today.

**Comment by ovflowd** [comment]
@bmuenzenmeyer could you re-review this?

**Review by bmuenzenmeyer — APPROVED** [review]
✅ `test`
✅ `lint`
✅ `build`
✅ `deploy`
✅ `serve`

**Comment by ovflowd** [comment]
I'm fast-tracking this PR to unblock related work.

**Pull request merged**
